app/src/main/java/nl/tudelft/b_b_w/model/block/GenesisBlock.java

```diff
@@ -27,11 +29,32 @@ public GenesisBlock(BlockData blockData) throws HashException {
         super(blockData);
 
         // verify integration and integrity
-        if (BuildConfig.DEBUG && blockData.getBlockType() != BlockType.GENESIS
-                || blockData.getSequenceNumber() != firstSequenceNumber
-                || !blockData.getPreviousHashChain().equals(notAvailable)
-                || !blockData.getPreviousHashSender().equals(notAvailable)) {
-            throw new AssertionError("Invalid genesis block " + this.toString());
+        if (BuildConfig.DEBUG) {
+            if (blockData.getBlockType() != BlockType.GENESIS
+                    || blockData.getSequenceNumber() != FIRST_SEQUENCE_NUMBER
+                    || !blockData.getPreviousHashChain().equals(notAvailable)
+                    || !blockData.getPreviousHashSender().equals(notAvailable)) {
+                throw new AssertionError("Invalid genesis block " + this.toString());
+            }
         }
     }
+
+    /**
+     * Easy method for genesis block creation.
+     * TODO: move this to constructor of genesis block
```

Comment:
You're saying this, but I can't find any issue or ticket which describes this. So please make one and don't assume it will just magicly happen.